### DISCUSSION TITLE
Allow addition models?

### DISCUSSION THREAD

**Discussion by wbste** [theredsix/cerebellum#1] [Ideas]
Any chance that a model like Molmo (that can "point") could be used as a self-host solution? I believe vLLM could be used as an endpoint?

https://molmo.allenai.org/blog

**Comment by theredsix** [comment]
I think we definitely would like to support local models via API on either vLLM or ollama in the future.

From a technical standpoint, the LLM needs to:
1. Understand the UI
2. Be able to plan actions that move the webpage state closer to the goal
3. Translate the actions from 2 to coordinates for mouse actions

From my experiments, all frontier models from GPT-4o to Gemini are capable of 1 & 2 out of the box, and even cost-optimized models like GPT-4o-mini can be fine-tuned to understand 1&2 as well. I think it's an open question whether the 72b version would work well in this format but it definitely be worth a shot. The barrier to running this experiment would be implementing a `vLLMPlanner`. Would you be interested in crafting `vLLMPlanner` using `AnthropicPlanner` as a guide?

**Reply by wbste** [reply]
That's likely above my skillset, but I'll take a look!

**Comment by theredsix** [comment]
<img width="1176" alt="Screenshot 2024-11-01 at 9 44 19 AM" src="https://github.com/user-attachments/assets/adfe67fa-370d-4c45-91ff-88dfbc95d816">

@wbste Molmo definitely does look promising.

**Reply by wbste** [reply]
Yeah molmo is the first open model I found that _should_ provide reliable coordinates for these sort of use cases. I also saw this..not sure if that opens more doors to other models or not: https://github.com/microsoft/OmniParser

**Comment by theredsix** [comment]
We're starting to gather data for a fine-tune of Molmo!

https://github.com/theredsix/cerebellum/tree/mind2web/training
